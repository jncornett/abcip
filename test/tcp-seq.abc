
a ( syn )
b ( syn, ack )
a ( ack )

# exact retrans
a ( ack, pay="ABCDEFGHIJ" )
a ( ack, pay="ABCDEFGHIJ", jump=-10 )
b ( ack )

# different data (and different cksum)
a ( ack, pay="0123456789" )
a ( ack, pay="0123456787", jump=-10 )
b ( ack )

# more data
a ( ack, pay="abcdefghij" )
a ( ack, pay="abcdefghijk", jump=-10 )
b ( ack )

# different seq
a ( ack, pay="9876543210" )
a ( ack, pay="9876543210", jump=-9 )
b ( ack )

# different cksum
a ( ack, pay="ZYXWVUTSRQ" )
a ( ack, pay="ZYXWVUTSRQ", jump=-10, cks=0x1234 )
b ( ack )

# same cksum - different data
a ( ack, pay="zyxwvutsrq" )
a ( ack, pay="HACKATTACK", jump=-10, cks=0x823F )
b ( ack )

a ( fin,ack )
b ( fin,ack )
a ( ack )

