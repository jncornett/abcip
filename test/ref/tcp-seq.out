IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [S], cksum 0xb3d9 (correct), seq 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0001 0000 4006 5cbc 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0008 0000 0001 0000 0000 5002  ..............P.
	0x0030:  2000 b3d9 0000                           ......
IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.8 > 10.1.2.3.48620: Flags [S.], cksum 0xb3c7 (correct), seq 1, ack 2, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0001 0000 4006 5cbc 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 0008 bdec 0000 0001 0000 0002 5012  ..............P.
	0x0030:  2000 b3c7 0000                           ......
IP (tos 0x0, ttl 64, id 2, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0xb3c8 (correct), ack 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0002 0000 4006 5cbb 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0008 0000 0002 0000 0002 5010  ..............P.
	0x0030:  2000 b3c8 0000                           ......
IP (tos 0x0, ttl 64, id 3, offset 0, flags [none], proto TCP (6), length 50)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0x595f (correct), seq 1:11, ack 1, win 8192, length 10
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0032 0003 0000 4006 5cb0 0a01 0203 0a09  .2....@.\.......
	0x0020:  0807 bdec 0008 0000 0002 0000 0002 5010  ..............P.
	0x0030:  2000 595f 0000 4142 4344 4546 4748 494a  ..Y_..ABCDEFGHIJ
IP (tos 0x0, ttl 64, id 4, offset 0, flags [none], proto TCP (6), length 50)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0x595f (correct), seq 1:11, ack 1, win 8192, length 10
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0032 0004 0000 4006 5caf 0a01 0203 0a09  .2....@.\.......
	0x0020:  0807 bdec 0008 0000 0002 0000 0002 5010  ..............P.
	0x0030:  2000 595f 0000 4142 4344 4546 4748 494a  ..Y_..ABCDEFGHIJ
IP (tos 0x0, ttl 64, id 2, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.8 > 10.1.2.3.48620: Flags [.], cksum 0xb3be (correct), ack 11, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0002 0000 4006 5cbb 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 0008 bdec 0000 0002 0000 000c 5010  ..............P.
	0x0030:  2000 b3be 0000                           ......
IP (tos 0x0, ttl 64, id 5, offset 0, flags [none], proto TCP (6), length 50)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0xaeaa (correct), seq 11:21, ack 1, win 8192, length 10
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0032 0005 0000 4006 5cae 0a01 0203 0a09  .2....@.\.......
	0x0020:  0807 bdec 0008 0000 000c 0000 0002 5010  ..............P.
	0x0030:  2000 aeaa 0000 3031 3233 3435 3637 3839  ......0123456789
IP (tos 0x0, ttl 64, id 6, offset 0, flags [none], proto TCP (6), length 50)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0xaeac (correct), seq 11:21, ack 1, win 8192, length 10
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0032 0006 0000 4006 5cad 0a01 0203 0a09  .2....@.\.......
	0x0020:  0807 bdec 0008 0000 000c 0000 0002 5010  ..............P.
	0x0030:  2000 aeac 0000 3031 3233 3435 3637 3837  ......0123456787
IP (tos 0x0, ttl 64, id 3, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.8 > 10.1.2.3.48620: Flags [.], cksum 0xb3b4 (correct), ack 21, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0003 0000 4006 5cba 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 0008 bdec 0000 0002 0000 0016 5010  ..............P.
	0x0030:  2000 b3b4 0000                           ......
IP (tos 0x0, ttl 64, id 7, offset 0, flags [none], proto TCP (6), length 50)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0xb8aa (correct), seq 21:31, ack 1, win 8192, length 10
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0032 0007 0000 4006 5cac 0a01 0203 0a09  .2....@.\.......
	0x0020:  0807 bdec 0008 0000 0016 0000 0002 5010  ..............P.
	0x0030:  2000 b8aa 0000 6162 6364 6566 6768 696a  ......abcdefghij
IP (tos 0x0, ttl 64, id 8, offset 0, flags [none], proto TCP (6), length 51)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0x4da9 (correct), seq 21:32, ack 1, win 8192, length 11
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0033 0008 0000 4006 5caa 0a01 0203 0a09  .3....@.\.......
	0x0020:  0807 bdec 0008 0000 0016 0000 0002 5010  ..............P.
	0x0030:  2000 4da9 0000 6162 6364 6566 6768 696a  ..M...abcdefghij
	0x0040:  6b                                       k
IP (tos 0x0, ttl 64, id 4, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.8 > 10.1.2.3.48620: Flags [.], cksum 0xb3a9 (correct), ack 32, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0004 0000 4006 5cb9 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 0008 bdec 0000 0002 0000 0021 5010  .............!P.
	0x0030:  2000 b3a9 0000                           ......
IP (tos 0x0, ttl 64, id 9, offset 0, flags [none], proto TCP (6), length 50)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0xa99a (correct), seq 32:42, ack 1, win 8192, length 10
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0032 0009 0000 4006 5caa 0a01 0203 0a09  .2....@.\.......
	0x0020:  0807 bdec 0008 0000 0021 0000 0002 5010  .........!....P.
	0x0030:  2000 a99a 0000 3938 3736 3534 3332 3130  ......9876543210
IP (tos 0x0, ttl 64, id 10, offset 0, flags [none], proto TCP (6), length 50)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0xa999 (correct), seq 33:43, ack 1, win 8192, length 10
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0032 000a 0000 4006 5ca9 0a01 0203 0a09  .2....@.\.......
	0x0020:  0807 bdec 0008 0000 0022 0000 0002 5010  ........."....P.
	0x0030:  2000 a999 0000 3938 3736 3534 3332 3130  ......9876543210
IP (tos 0x0, ttl 64, id 5, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.8 > 10.1.2.3.48620: Flags [.], cksum 0xb39e (correct), ack 43, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0005 0000 4006 5cb8 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 0008 bdec 0000 0002 0000 002c 5010  .............,P.
	0x0030:  2000 b39e 0000                           ......
IP (tos 0x0, ttl 64, id 11, offset 0, flags [none], proto TCP (6), length 50)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0x03ea (correct), seq 43:53, ack 1, win 8192, length 10
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0032 000b 0000 4006 5ca8 0a01 0203 0a09  .2....@.\.......
	0x0020:  0807 bdec 0008 0000 002c 0000 0002 5010  .........,....P.
	0x0030:  2000 03ea 0000 5a59 5857 5655 5453 5251  ......ZYXWVUTSRQ
IP (tos 0x0, ttl 64, id 12, offset 0, flags [none], proto TCP (6), length 50)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0x1234 (incorrect -> 0x03ea), seq 43:53, ack 1, win 8192, length 10
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0032 000c 0000 4006 5ca7 0a01 0203 0a09  .2....@.\.......
	0x0020:  0807 bdec 0008 0000 002c 0000 0002 5010  .........,....P.
	0x0030:  2000 1234 0000 5a59 5857 5655 5453 5251  ...4..ZYXWVUTSRQ
IP (tos 0x0, ttl 64, id 6, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.8 > 10.1.2.3.48620: Flags [.], cksum 0xb394 (correct), ack 53, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0006 0000 4006 5cb7 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 0008 bdec 0000 0002 0000 0036 5010  .............6P.
	0x0030:  2000 b394 0000                           ......
IP (tos 0x0, ttl 64, id 13, offset 0, flags [none], proto TCP (6), length 50)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0x633f (correct), seq 53:63, ack 1, win 8192, length 10
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0032 000d 0000 4006 5ca6 0a01 0203 0a09  .2....@.\.......
	0x0020:  0807 bdec 0008 0000 0036 0000 0002 5010  .........6....P.
	0x0030:  2000 633f 0000 7a79 7877 7675 7473 7271  ..c?..zyxwvutsrq
IP (tos 0x0, ttl 64, id 14, offset 0, flags [none], proto TCP (6), length 50)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0x823f (incorrect -> 0x4f1d), seq 53:63, ack 1, win 8192, length 10
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0032 000e 0000 4006 5ca5 0a01 0203 0a09  .2....@.\.......
	0x0020:  0807 bdec 0008 0000 0036 0000 0002 5010  .........6....P.
	0x0030:  2000 823f 0000 4841 434b 4154 5441 434b  ...?..HACKATTACK
IP (tos 0x0, ttl 64, id 7, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.8 > 10.1.2.3.48620: Flags [.], cksum 0xb38a (correct), ack 63, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0007 0000 4006 5cb6 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 0008 bdec 0000 0002 0000 0040 5010  .............@P.
	0x0030:  2000 b38a 0000                           ......
IP (tos 0x0, ttl 64, id 15, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [F.], cksum 0xb389 (correct), seq 63, ack 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 000f 0000 4006 5cae 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0008 0000 0040 0000 0002 5011  .........@....P.
	0x0030:  2000 b389 0000                           ......
IP (tos 0x0, ttl 64, id 8, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.8 > 10.1.2.3.48620: Flags [F.], cksum 0xb388 (correct), seq 1, ack 64, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0008 0000 4006 5cb5 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 0008 bdec 0000 0002 0000 0041 5011  .............AP.
	0x0030:  2000 b388 0000                           ......
IP (tos 0x0, ttl 64, id 16, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0xb388 (correct), ack 2, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0010 0000 4006 5cad 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0008 0000 0041 0000 0003 5010  .........A....P.
	0x0030:  2000 b388 0000                           ......
